src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,28 +111,38 @@ data CCurrency
     deriving (Show, Read, Generic)
 
 -- | Client hash
-newtype CHash = CHash Text deriving (Show, Eq, Generic, Buildable)
+newtype CHash = CHash Text
+    deriving (Show, Eq, Generic, Buildable)
 
 instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash
```

Comment:
@Martoon-00 The code is working, so it's not really necessary, but I was curious to look why would someone use such a thing. It's an observation that if anyone uses them, **they should be safe**, nothing else. If ADT were used, the code would be safe, now it's safe as _long as you know what you are doing in the code_. 
And I don't know about you, but I rarely know what I'm doing in the code :smile: